### PR TITLE
Thunks/gen: Add support for compiling against clang 19

### PR DESCRIPTION
**Pull request by WhatAmISupposedToPutHere**

### CODE REVIEW

**Review by alyssarosenzweig — APPROVED** [review]
nice cleanup :)

**Review by Sonicadvance1 — APPROVED** [review]

**Comment by neobrain** [comment]
Unit tests (`BUILD_TESTS=ON`) will still fail to compile with LLVM 19:

```diff
diff --git unittests/ThunkLibs/generator.cpp unittests/ThunkLibs/generator.cpp
index d3fc9c7c0..38eadbfd6 100644
--- unittests/ThunkLibs/generator.cpp
+++ unittests/ThunkLibs/generator.cpp
@@ -417,7 +417,7 @@ TEST_CASE_METHOD(Fixture, "VersionedLibrary") {

   CHECK_THAT(output, matches(callExpr(callee(functionDecl(hasName("dlopen"))), hasArgument(0, stringLiteral().bind("libname"))))
                        .check_binding(
-                         "libname", +[](const clang::StringLiteral* lit) { return lit->getString().endswith(".so.123"); }));
+                         "libname", +[](const clang::StringLiteral* lit) { return lit->getString().ends_with(".so.123"); }));
 }

 TEST_CASE_METHOD(Fixture, "FunctionPointerViaType") {
```

Other than that, tested and confirmed to produce working error messages equivalent to older LLVM versions. Well done!

File: unittests/ThunkLibs/generator.cpp [hunk 1]
```diff
@@ -415,9 +415,17 @@ TEST_CASE_METHOD(Fixture, "UnknownAnnotation") {
...
+#if CLANG_VERSION_MAJOR >= 19
+#define ENDSWITH ends_with
+#else
+#define ENDSWITH endswith
+#endif
+
   CHECK_THAT(output, matches(callExpr(callee(functionDecl(hasName("dlopen"))), hasArgument(0, stringLiteral().bind("libname"))))
                        .check_binding(
-                         "libname", +[](const clang::StringLiteral* lit) { return lit->getString().endswith(".so.123"); }));
+                         "libname", +[](const clang::StringLiteral* lit) { return lit->getString().ENDSWITH(".so.123"); }));
```
Comment (neobrain) [inline unittests/ThunkLibs/generator.cpp:426]: Could you just wrap the line itself in the ifdef, or disable the VersionedLibrary test entirely? Using a macro to expand to a member function name seems unorthodox and can easily be avoided that way.

File: unittests/ThunkLibs/generator.cpp [hunk 2]
```diff
@@ -415,9 +415,15 @@ TEST_CASE_METHOD(Fixture, "UnknownAnnotation") {
 TEST_CASE_METHOD(Fixture, "VersionedLibrary") {
   const auto output = run_thunkgen_host("", "template<auto> struct fex_gen_config { int version = 123; };\n");
 
+#if CLANG_VERSION_MAJOR >= 19
```
Comment (neobrain) [inline unittests/ThunkLibs/generator.cpp:418]: The renamed API was added in LLVM 17 (in addition to deprecating the one they removed in LLVM 19).

```suggestion
#if CLANG_VERSION_MAJOR >= 17
```

**Review by neobrain — APPROVED** [review]
Perfect, thanks again!

**Pull request merged**